pysteps/blending/steps.py

```diff
@@ -1451,16 +1455,51 @@ def worker(j):
                         # forecast outside the radar domain. Therefore, fill these
                         # areas with the "..._mod_only" blended forecasts, consisting
                         # of the NWP and noise components.
-                        nan_indices = np.isnan(R_f_new)
-                        R_f_new[nan_indices] = R_f_new_mod_only[nan_indices]
-                        nan_indices = np.isnan(R_pm_blended)
-                        R_pm_blended[nan_indices] = R_pm_blended_mod_only[nan_indices]
-                        # Finally, fill the remaining nan values, if present, with
-                        # the minimum value in the forecast
-                        nan_indices = np.isnan(R_f_new)
-                        R_f_new[nan_indices] = np.nanmin(R_f_new)
-                        nan_indices = np.isnan(R_pm_blended)
-                        R_pm_blended[nan_indices] = np.nanmin(R_pm_blended)
+                        if create_smooth_radar_mask is True:
+                            # Compute the smooth dilated mask
+                            nan_indices = np.isnan(R_f_new)
```

Comment:
I would not replicate in the if..else.. block any line that is common to both cases, as for example the this line which could go before the if statement

Reply:
This is fixed in the new version